src/test/cypress/support/requests/CourseManagementRequests.ts

```diff
@@ -70,6 +79,10 @@ export class CourseManagementRequests {
         programmingTemplate.dueDate = dueDate.toISOString();
         programmingTemplate.course = course;
 
+        if (!skipGroupSynchronization) {
```

Comment:
couldn't we add `const runsOnBamboo: boolean = Cypress.env('isBamboo');` in this file and add it to the if statement: `if (!skipGroupSynchronization && runsOnBamboo) { cy.waitForGroupSynchronization(); }`
Then, when we enter the branch, we actually wait, as indicated by the method name.

Reply:
Yeah we could do that. What would you prefer? Should I move the `if (runsOnBamboo)` outside of the `waitForGroupSynchronization` command (then we have to put the if in every calling code segment) or should I rename the command to `waitForGroupSynchronizationIfRequired`?

Reply:
If a reorder the tests in ProgrammingExercise.spec.ts I could get rid of the `skipGroupSynchronization` flag in total because this is the only test which currently creates more than one programming exercise per course. I added this code mostly for future tests, but we might not even have tests which create multiple programming exercises

Reply:
i'd prefer the `runsOnBamboo` check before and not within the cypress command, so exactly how you refactored it now. If you agree, that the command is still practical to use, it's fine.